app/utils/purl.js

```diff
@@ -0,0 +1,22 @@
+import window from 'ember-window-mock';
+
+/**
+ * Adds a repository_url query parameter to a PURL string based on the host.
+ *
+ * @param {string} purl - The base PURL string (e.g., "pkg:cargo/serde@1.0.0")
+ * @param {string} [host] - The host to use for repository URL. Defaults to current window location host.
+ * @returns {string} The PURL with repository_url parameter added, or unchanged if host is crates.io
+ */
+export function addRegistryUrl(purl) {
+  let host = window.location.host;
+
+  // Don't add repository_url for the main crates.io registry
+  if (host === 'crates.io') {
+    return purl;
+  }
+
+  // Add repository_url query parameter
+  const repositoryUrl = `https://${host}/`;
+  const separator = purl.includes('?') ? '&' : '?';
```

Comment:
It feels like this is redundant, since we can't ever generate a package URL that includes a `?` right now?

(I guess, more broadly, I'm not quite clear on why this is separated from the basic PURL generation in the model.)

Reply:
> I'm not quite clear on why this is separated from the basic PURL generation in the model

I initially implemented the same property on the `crate` model too and wanted to DRY it up 😉